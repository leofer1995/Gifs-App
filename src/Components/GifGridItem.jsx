import React from "react";
import PropTypes from 'prop-types';



const GifGridItem = ({title, url})=>{
   
    return (
        <div className='Card animate__animated animate__backInLeft'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem;

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}