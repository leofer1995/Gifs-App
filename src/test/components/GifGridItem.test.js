import GifGridItem from "../../Components/GifGridItem";
import {shallow} from 'enzyme';


describe('Pruebas en el componente GifGridItem',()=>{

    const url = 'https://localhost/hola.jpg';
    const title = 'Hola Mundo';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('Prueba de renderizado del componente',()=>{
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title',()=>{ 
        const p =  wrapper.find('p');
        expect(p.text().trim()).toBe(title)
    });

    test('debe tener la imagen igual al url y alt de los props', ()=>{

        const img = wrapper.find('img');
        //console.log(img.props().src)
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);

    });

    test('debe tener como clase animate__backInLeft',()=>{
        const div = wrapper.find('div');
        //console.log(div.props())
        const className = div.props().className
        expect(className.includes('animate__backInLeft')).toBe(true)
    });
})
