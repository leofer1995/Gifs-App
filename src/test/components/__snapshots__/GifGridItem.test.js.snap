// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Prueba de renderizado del componente 1`] = `
<div
  className="Card animate__animated animate__backInLeft"
>
  <img
    alt="Hola Mundo"
    src="https://localhost/hola.jpg"
  />
  <p>
    Hola Mundo
  </p>
</div>
`;
